assets/queries/dockerfile/pin_versions_apt_get_install/query.rego

```diff
@@ -1,31 +1,54 @@
 package Cx
 
+import data.generic.dockerfile as dockerLib
+
 CxPolicy[result] {
 	resource := input.document[i].command[name][_]
 	resource.Cmd == "run"
-	command := resource.Value[0]
 
-	contains(resource.Value[0], "apt-get install")
-	commandSplited := split(resource.Value[0], " ")
+	count(resource.Value) == 1
+	commands := resource.Value[0]
 
-	"install" == commandSplited[index]
+	aptGet := regex.find_n("apt-get (-(-)?[a-zA-Z]+ *)*install", commands, -1)
+	aptGet != null
 
-	index + 1 <= count(commandSplited)
-	packages := array.slice(commandSplited, index + 1, count(commandSplited))
+	packages = dockerLib.getPackages(commands, aptGet)
+	regex.match("^[a-zA-Z]", packages[j]) == true
 
-	some pack
-	not startswith(packages[pack], "-")
-	noVersion(packages[pack])
+	not dockerLib.withVersion(packages[j])
 
 	result := {
 		"documentId": input.document[i].id,
-		"searchKey": sprintf("FROM={{%s}}.RUN={{%s}}", [name, resource.Value[0]]),
+		"searchKey": sprintf("FROM={{%s}}.RUN={{%s}}", [name, commands]),
 		"issueType": "MissingAttribute",
-		"keyExpectedValue": "When installing a package, its pin version should be defined",
-		"keyActualValue": "The pin version is not defined",
+		"keyExpectedValue": sprintf("Package '%s' has version defined", [packages[j]]),
+		"keyActualValue": sprintf("Package '%s' does not have version defined", [packages[j]]),
+	}
+}
+
+CxPolicy[result] {
+	resource := input.document[i].command[name][_]
+	resource.Cmd == "run"
+
+	count(resource.Value) > 1
+
+	isAptGet(resource.Value)
+
+	resource.Value[j] != "install"
+	resource.Value[j] != "apt-get"
+	regex.match("^[a-zA-Z]", resource.Value[j]) == true
+	not dockerLib.withVersion(resource.Value[j])
+
+	result := {
+		"documentId": input.document[i].id,
+		"searchKey": sprintf("FROM={{%s}}.{{%s}}", [name, resource.Original]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": sprintf("Package '%s' has version defined", [resource.Value[j]]),
+		"keyActualValue": sprintf("Package '%s' does not have version defined", [resource.Value[j]]),
 	}
 }
 
-noVersion(pack) {
-	not contains(pack, "=")
+isAptGet(command) {
+	contains(command[x], "apt-get")
+	contains(command[j], "install")
```

Comment:
Perhaps its a good idea to establish a relation between *x* and *j*

Reply:
It is to verify if different indexes have "apt-get" and "install".
For example, RUN ["apt-get", "install", "blunder"]